### PR TITLE
[Refactor] Rename `push_dir` to `chdir`

### PR DESCRIPTION
**Pull request by ybiquitous**
The name `push_dir` is confusing to me because it looks that the word "push" has some side-effect.
(Please consider `Array#push`)

So, I suggest to rename it to `chdir` to improve the readability.
The standard library has the same name:
- [`Dir.chdir`](https://ruby-doc.org/core-2.7.1/Dir.html#method-c-chdir)
- [`Process.spawn`](https://ruby-doc.org/core-2.7.1/Process.html#method-c-spawn)

See also <https://github.com/sider/runners/pull/1084#discussion_r425560235>

### CODE REVIEW

**Comment by ybiquitous** [comment]
> [IMO] Although I don't disagree with adding `alias`, the name of `push_dir` is appropriate for the current implementation since it has a "stack." So, it might be good to delete `dir_stack` if you try to implement it.

https://github.com/sider/runners/pull/1084/files#r425571600 by @yykamei

**Comment by ybiquitous** [comment]
Thank you for the feedback!

Honestly, I don't want to use the word "stack" because it's an implementation detail and what this module's caller doesn't need to know. 🤔

Also, I think the word "push" is not accurate because this method does both **push** and **pop**.

**Comment by yykamei** [comment]
> Honestly, I don't want to use the word "stack" because it's an implementation detail and what this module's caller doesn't need to know. 
> 
> Also, I think the word "push" is not accurate because this method does both **push** and **pop**.

I agree with you. As I said, `dir_stack` is unnecessary for `Shell`. If `Shell` has to handle many directories and store them, the stack data structure is useful, but our Runners do not require so many directories. Instead, Runners require only the feature like `pushd` and `popd` of Bash. [`Dir.chdir`](https://ruby-doc.org/core-2.7.1/Dir.html#method-c-chdir) satisfies the feature.

**Comment by ybiquitous** [comment]
Make sense. Surely, we don't need a stack. Here is enough:

```ruby
class Shell
  def chdir(dir)
    Dir.chdir(dir) do |path|
      yield Pathname(path)
    end
  end
end
```

I will try it if you are not against it.

File: lib/runners/shell.rb [hunk 1]
```diff
@@ -36,26 +36,28 @@ def bugsnag_meta_data
...
     def initialize(current_dir:, trace_writer:, env_hash:)
       @trace_writer = trace_writer
-      @dir_stack = [current_dir]
+      @current_dir = current_dir
       @env_hash_stack = [env_hash]
     end
 
     def current_dir
-      dir_stack.last or raise "Empty dir stack"
+      @current_dir
```
[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/runners/shell.rb]: Make sense. Fixed via f466457.

File: test/shell_test.rb [hunk 2]
```diff
@@ -16,6 +16,32 @@ def assert_trace_writer(expected)
...
+      ret = shell.chdir(path / "foo") do |dir|
+        assert_equal(path / "foo", dir)
+        assert_equal(path / "foo", shell.current_dir)
+
+        # Nested
+        (dir / "bar").tap(&:mkpath)
+        shell.chdir(dir / "bar") do |dir2|
+          assert_equal(dir / "bar", dir2)
+          assert_equal(dir / "bar", shell.current_dir)
+        end
```
Comment (ybiquitous) [inline test/shell_test.rb:34]: I've removed the stack structure but the same thing is realized!

**Comment by ybiquitous** [comment]
Thank you so much! 🥰

**Pull request merged**